### PR TITLE
python3Packages.nix-prefetch-github: 6.0.1 -> 7.0.0

### PR DESCRIPTION
**Pull request by seppeljordan**
###### Description of changes

nix-prefetch-github was update to version 7.0.0.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [x] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by tjni** [comment]
Looks good! There are a few dozen usages of `nix-prefetch-github` in update scripts -- could those be updated in this PR to use the new `hash` field?

**Comment by seppeljordan** [comment]
Do I need to do anything?

**Comment by tjni** [comment]
I added some commits to do this in https://github.com/seppeljordan/nixpkgs/pull/1. Can you review and merge into this PR if it looks good to you?

File: pkgs/servers/web-apps/lemmy/pin.json [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {
   "version": "0.18.1",
-  "serverSha256": "sha256-jYbrbIRyXo2G113ReG32oZ56ed2FEB/ZBcqYAxoxzGQ=",
-  "serverCargoSha256": "sha256-7DNMNPSjzYY45DlR6Eo2q6QdwrMrRb51cFOnXfOuub0=",
-  "uiSha256": "sha256-tc7fGA4okIv+3kq5t6I+EN+owdekCgAdk0EtkDgodIU=",
-  "uiYarnDepsSha256": "sha256-/T1PQ0XsGY82/RlLq94hJgv8Yx1bIf2gcklhgtPu1RM="
+  "serverHash": "sha256-jYbrbIRyXo2G113ReG32oZ56ed2FEB/ZBcqYAxoxzGQ=",
+  "serverCargoHash": "sha256-7DNMNPSjzYY45DlR6Eo2q6QdwrMrRb51cFOnXfOuub0=",
+  "uiHash": "sha256-tc7fGA4okIv+3kq5t6I+EN+owdekCgAdk0EtkDgodIU=",
+  "uiYarnDepshash": "sha256-/T1PQ0XsGY82/RlLq94hJgv8Yx1bIf2gcklhgtPu1RM="
```
Comment (tjni) [inline pkgs/servers/web-apps/lemmy/pin.json:6]: Just a small typo here: `uiYarnDepshash` => `uiYarnDepsHash`.

**Review by tjni — COMMENTED** [review]
Thank you for merging everything!

File: pkgs/servers/web-apps/lemmy/pin.json (hunk 1 above)
Comment (seppeljordan) [reply pkgs/servers/web-apps/lemmy/pin.json]: Thanks. I fixed it.

**Comment by tjni** [comment]
We're a bit unlucky: https://github.com/NixOS/nixpkgs/pull/242130 introduced a conflict as well.

**Comment by seppeljordan** [comment]
@tjni I addressed the conflict.

**Comment by tjni** [comment]
Result of `nixpkgs-review pr 242092` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package marked as broken and skipped:</summary>
  <ul>
    <li>solana-validator</li>
  </ul>
</details>
<details>
  <summary>20 packages built:</summary>
  <ul>
    <li>element-desktop</li>
    <li>element-desktop-wayland</li>
    <li>emacsPackages.evil-textobj-tree-sitter</li>
    <li>emacsPackages.evil-tree-edit</li>
    <li>emacsPackages.helm-tree-sitter</li>
    <li>emacsPackages.pygn-mode</li>
    <li>emacsPackages.symex</li>
    <li>emacsPackages.tree-sitter</li>
    <li>emacsPackages.tree-sitter-ess-r</li>
    <li>emacsPackages.tree-sitter-indent</li>
    <li>emacsPackages.tree-sitter-ispell</li>
    <li>emacsPackages.tree-sitter-langs</li>
    <li>emacsPackages.tsc</li>
    <li>lemmy-server</li>
    <li>nix-prefetch-github (python310Packages.nix-prefetch-github)</li>
    <li>nix-prefetch-github.dist (python310Packages.nix-prefetch-github.dist)</li>
    <li>python311Packages.nix-prefetch-github</li>
    <li>python311Packages.nix-prefetch-github.dist</li>
    <li>schildichat-desktop</li>
    <li>schildichat-desktop-wayland</li>
  </ul>
</details>

**Comment by tjni** [comment]
We have another (more complicated) merge conflict in https://github.com/NixOS/nixpkgs/pull/242979/. Do you mind if I take over this PR? It is taking awhile to get committed, and I feel bad that you have to deal with all of these conflicts related to what I've asked you to push.

**Comment by seppeljordan** [comment]
@tjni sure, go ahead.

**Comment by tjni** [comment]
Continued in https://github.com/NixOS/nixpkgs/pull/243018.

**Pull request closed without merging**